src/components/EarnDashboard/components/StakingPositions/LpPositions.tsx

```diff
@@ -0,0 +1,189 @@
+import { Button, Flex, Stack } from '@chakra-ui/react'
+import { Tag } from '@chakra-ui/tag'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { cosmosAssetId, cosmosChainId, fromAssetId, osmosisChainId } from '@shapeshiftoss/caip'
+import { DefiAction } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import qs from 'qs'
+import { useCallback, useMemo } from 'react'
+import { useHistory, useLocation } from 'react-router'
+import type { Column, Row } from 'react-table'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { ReactTable } from 'components/ReactTable/ReactTable'
+import { RawText } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { LpEarnOpportunityType, OpportunityId } from 'state/slices/opportunitiesSlice/types'
+import { getUnderlyingAssetIdsBalances } from 'state/slices/opportunitiesSlice/utils'
+import {
+  selectAggregatedEarnUserLpOpportunities,
+  selectAssets,
+  selectCryptoMarketData,
+  selectFirstAccountIdByChainId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type ProviderPositionProps = {
+  ids: OpportunityId[]
+  assetId: AssetId
+}
+
+export type RowProps = Row<LpEarnOpportunityType>
+
+export const LpPositions: React.FC<ProviderPositionProps> = ({ ids, assetId }) => {
+  const location = useLocation()
+  const history = useHistory()
+  const {
+    state: { isConnected, isDemoWallet },
+    dispatch,
+  } = useWallet()
+  const assets = useAppSelector(selectAssets)
+  const marketData = useAppSelector(selectCryptoMarketData)
+  const lpOpportunities = useAppSelector(selectAggregatedEarnUserLpOpportunities)
+  const filteredDown = lpOpportunities.filter(e => ids.includes(e.assetId as OpportunityId))
+
+  const cosmosAccountId = useAppSelector(state =>
+    selectFirstAccountIdByChainId(state, cosmosChainId),
+  )
+  const osmosisAccountId = useAppSelector(state =>
+    selectFirstAccountIdByChainId(state, osmosisChainId),
+  )
```

Comment:
??? why?

Reply:
@gomesalexandre are you able to suggest a way to avoid this? this is not going to generalize

Reply:
https://github.com/shapeshift/web/pull/3747/files#r1120664455 these two can be removed